integration-tests/cstor-pool/cstor_integration_test.go

```diff
@@ -0,0 +1,370 @@
+		It("a new pool should come up again", func() {
+			clientInstance.deleteCSP(spcObj.Name, 1)
+			// We expect 3 cstorPool objects.
+			cspCount := isHealthyCspCount(spcObj.Name, 3)
+			Expect(cspCount).To(Equal(3))
+		})
+	})
+
+	//Test Case #3 : Dependent on above test case #1 . | TestType : Reconciliation
+	When("We delete 2 pool out of 3 by deleting 2 csps", func() {
+		It("2 new pool should come up again", func() {
+			clientInstance.deleteCSP(spcObj.Name, 2)
+			// We expect 3 cstorPool objects.
+			cspCount := isHealthyCspCount(spcObj.Name, 3)
+			Expect(cspCount).To(Equal(3))
+		})
+	})
+
+	//Test Case #3 : Dependent on above test case #1 . | TestType : Reconciliation
+	When("We delete 3 pool out of 3 by deleting 3 csps", func() {
+		It("3 new pool should come up again", func() {
+			clientInstance.deleteCSP(spcObj.Name, 3)
+			// We expect 3 cstorPool objects.
+			cspCount := isHealthyCspCount(spcObj.Name, 3)
+			Expect(cspCount).To(Equal(3))
+		})
+	})
+
+})
+
+
+var _ = AfterSuite(func() {
+	spcClient, err := spc.KubeClient().WithFlag(KubeConfigPath)
+	spcList, err := spcClient.List(metav1.ListOptions{})
+	Expect(err).To(BeNil())
+	for _, spc := range spcList.Items {
+		_, err := spcClient.Delete(spc.Name, &metav1.DeleteOptions{})
+		Expect(err).To(BeNil())
+	}
+})
+
+var _ = BeforeSuite(func() {
+	for _, f := range clientBuilderFuncList {
+		f()
+	}
+})
+
+// clientInstacne is the initialized empty instance of client type.
+var clientInstance = &client{}
+
+type clientBuilderFunc func() *client
+
+var clientBuilderFuncList = []clientBuilderFunc{
+	clientInstance.newSpcClient,
+	clientInstance.newCspClient,
+}
+
+func (c *client) newSpcClient() *client {
+	newCspClient, err := csp.KubeClient().WithFlag(KubeConfigPath)
+	Expect(err).To(BeNil())
+	c.cspClient = newCspClient
+	return c
+}
+
+func (c *client) newCspClient() *client {
+	newSpcClient, err := spc.KubeClient().WithFlag(KubeConfigPath)
+	Expect(err).To(BeNil())
+	c.spcClient = newSpcClient
+	return c
+}
+
+func (c *client) getHealthyCSPCount(spcName string) int {
+	cspApiList, err := c.cspClient.List(metav1.ListOptions{})
+	Expect(err).To(BeNil())
+	count := csp.
+		ListBuilderForApiList(cspApiList).
+		List().
+		Filter(csp.HasLabel(string(apis.StoragePoolClaimCPK), spcName), csp.IsStatus("Healthy")).Len()
+	return count
+}
+
+func (c *client) deleteCSP(spcName string, deleteCount int) {
+	cspApiList, err := c.cspClient.List(metav1.ListOptions{})
+	Expect(err).To(BeNil())
+	cspList := csp.
+		ListBuilderForApiList(cspApiList).
+		List().
+		Filter(csp.HasLabel(string(apis.StoragePoolClaimCPK), spcName), csp.IsStatus("Healthy"))
+	cspCount := cspList.Len()
+	Expect(deleteCount).Should(BeNumerically("<=", cspCount))
+
+	for i := 0; i < deleteCount; i++ {
+		_, err := clientInstance.cspClient.Delete(cspList.Items[i].Object.Name, &metav1.DeleteOptions{})
+		Expect(err).To(BeNil())
+
+	}
+}
+
+func isHealthyCspCount(spcName string, expectedCspCount int) int {
```

Comment:
Can we have this as a method to `operation`

Reply:
We need not bind it. If we bind it again at the caller side we will always need to instantiate the object.
This is just a function that abstracts these logic from BDD block making it easier for readers.

Reply:
But now the cientInstance is available globally -- your comment make sense to me.
Will fix it.